src/Client.php

```diff
@@ -1186,6 +1186,39 @@ public function imageSearch($term)
         )->then(array($this->parser, 'expectJson'));
     }
 
+    /**
+     * Create a new image from a container
+     *
+     * @param array  $config    The container configuration (body of the request)
+     * @param string $container The ID or name of the container to commit
+     * @param string $repo      Repository name for the created image
+     * @param string $tag       Tag name for the created image
+     * @param string $comment   Commit message
+     * @param string $author    Author of the image (e.g., John Hannibal Smith <hannibal@a-team.com>)
+     * @param bool   $pause     Whether to pause the container before committing, default: true
+     * @param string $changes   Dockerfile instructions to apply while committing
+     * @return PromiseInterface
+     * @link https://docs.docker.com/engine/api/v1.41/#operation/ImageCommit
+     */
+    public function commit($config, $container, $repo = null, $tag = null, $comment = null, $author = null, $pause = true, $changes = null)
```

Comment:
Are you aware of any other API implementations of this interface? I wonder if it makes sense to perhaps name this `containerCommit()` to be more in line with the existing API methods?

Also, are you aware of any other implementations that take advantage of the `$config` array? I wonder if we should prominently feature the `$config`, use the `$container` as first argument or even drop a number of these parameters?

Reply:
Hey @dinooo13, thanks for your work on this so far :+1: 

@clue and I tried this out using docker one the command line and we looked at the necessary parameters when committing a container. It seems like the container parameter is the most important one in there followed by the repository and tag parameters (nearly the same as your suggestion).
We didn't find a config parameter tho, so I'm wondering what your use case for this is, to get an better understanding on why you added the `$config` parameter.

Reply:
Hey @SimonFrings I found this in the Docke Engine API docs.
https://docs.docker.com/engine/api/v1.41/#operation/ImageCommit
I'm not a Docker expert by any means, so I just oriented myself to it, maybe it's for configuration on top of the used container. But I guess YAGNI and should I update the PR to be more inline with the command line order of params?

Reply:
You're right, they implement it too, the only difference is that they put the `$config` parameter at the end of the parameter list. I'm not the one too judge if this is needed or not so I'm okay with adding this as the last parameter.

What's your opinion on this?

Reply:
I will update the PR so it will be the last parameter. Thats a good solution imo.

Reply:
PR is updated 😄

Reply:
It's squashed :shipit: